js/model3D/shared/Canvas3D.svelte

```diff
@@ -52,176 +42,91 @@
-
-		scene.createDefaultCameraOrLight();
-		scene.useRightHandedSystem = true;
-		scene.clearColor = scene.clearColor = new BABYLON.Color4(...clear_color);
-
-		engine.runRenderLoop(() => {
-			scene.render();
+		// Initialize BabylonJS viewer
+		BABYLON_VIEWER.createViewerForCanvas(canvas, {
+			clearColor: clear_color,
+			useRightHandedSystem: true,
+			cameraAutoOrbit: { enabled: false },
+			onInitialized: (details: Readonly<BABYLON_VIEWER.ViewerDetails>) => {
+				viewerDetails = details;
+			}
+		}).then((promiseViewer: BABYLON_VIEWER.Viewer) => {
+			viewer = promiseViewer;
+			mounted = true;
 		});
 
-		function onWindowResize(): void {
-			engine.resize();
-		}
-		window.addEventListener("resize", onWindowResize);
-
-		mounted = true;
-
 		return () => {
-			scene.dispose();
-			engine.dispose();
-			window.removeEventListener("resize", onWindowResize);
+			viewer.dispose();
 		};
 	});
 
 	$: mounted && load_model(resolved_url);
 
-	function load_model(url: string | undefined): void {
-		if (scene) {
-			// Dispose of the previous model before loading a new one
-			scene.meshes.forEach((mesh) => {
-				mesh.dispose();
-			});
-
-			if (point_cloud_system) {
-				point_cloud_system.dispose();
-				point_cloud_system = null;
-			}
+	function setRenderingMode(pointsCloud: boolean, wireframe: boolean): void {
+		viewerDetails.scene.forcePointsCloud = pointsCloud;
+		viewerDetails.scene.forceWireframe = wireframe;
+	}
 
-			// Load the new model
-			if (url) {
-				BABYLON.SceneLoader.ShowLoadingScreen = false;
-				BABYLON.SceneLoader.Append(
-					url,
-					"",
-					scene,
-					() => {
-						if (display_mode === "point_cloud") {
-							create_point_cloud(scene);
-						} else if (display_mode === "wireframe") {
-							create_wireframe(scene);
-						} else {
-							create_camera(scene, camera_position, zoom_speed, pan_speed);
+	function load_model(url: string | undefined): void {
+		if (viewer) {
+			viewer
+				.loadModel(url!, {
+					pluginOptions: {
+						obj: {
+							importVertexColors: true
 						}
-					},
-					undefined,
-					undefined,
-					"." + value.path.split(".").pop()
-				);
-			}
+					}
+				})
+				.then(() => {
+					if (display_mode === "point_cloud") {
+						setRenderingMode(true, false);
+					} else if (display_mode === "wireframe") {
+						setRenderingMode(false, true);
+					} else {
+						update_camera(camera_position, zoom_speed, pan_speed);
+					}
+				});
 		}
 	}
 
-	function create_camera(
-		scene: BABYLON.Scene,
+	function update_camera(
 		camera_position: [number | null, number | null, number | null],
 		zoom_speed: number,
 		pan_speed: number
 	): void {
-		scene.createDefaultCamera(true, true, true);
-		var helperCamera = scene.activeCamera! as BABYLON.ArcRotateCamera;
+		const camera = viewerDetails.camera;
 		if (camera_position[0] !== null) {
-			helperCamera.alpha = BABYLON.Tools.ToRadians(camera_position[0]);
+			camera.alpha = BABYLON.Tools.ToRadians(camera_position[0]);
```

Comment:
Isn't `BABYLON` gone now (this was the previous reference to the UMD module)? Maybe this is also why the camera reset didn't work for @dawoodkhan82 as mentioned here: https://github.com/gradio-app/gradio/pull/10847#issuecomment-2744635324

Seems like instead we should `import { Tools } from "@babylonjs/core/Misc/tools"`. Alternatively, you could just change this to:
```suggestion
			camera.alpha = (camera_position[0] * Math.PI) / 180;
```
Then you don't need to pull in the whole `Tools` class just for this, and keep the bundle size smaller.